ui/apps/platform/cypress/integration/certexpiration.test.js

```diff
@@ -1,125 +1,175 @@
-import withAuth from '../helpers/basicAuth';
+
 import * as api from '../constants/apiEndpoints';
 import { selectors } from '../constants/CertExpiration';
-import { visitMainDashboard } from '../helpers/main';
+import { systemConfigUrl } from '../constants/SystemConfigPage';
+import withAuth from '../helpers/basicAuth';
+import {
+    credentialexpiryCentralAlias,
+    credentialexpiryScannerAlias,
+    mypermissionsAlias,
+    visitWithGenericResponses,
+} from '../helpers/visit';
+
+/*
+ * Visit System Configuration page for certificate expiration tests:
+ * It has a minimal number of page-specific requests.
+ * The credentialexpiry and mypermissions requests are generic for any page.
+ */
+function visitForCredentialExpiry(staticResponseMapGeneric) {
+    visitWithGenericResponses(systemConfigUrl, staticResponseMapGeneric, {
+        config: {
+            method: 'GET',
+            url: api.system.config,
+        },
+    });
+}
 
 describe('Cert Expiration Banner', () => {
     withAuth();
 
-    const mockCertExpiryAndVisitHomepage = (endpoint, expiry) => {
-        cy.intercept('GET', endpoint, {
-            body: { expiry },
-        }).as('certExpiry');
-        visitMainDashboard();
-        cy.wait('@certExpiry');
-    };
-
     describe('Central', () => {
         it('should not display banner if central cert is expiring more than 14 days later', () => {
             const expiry = dateFns.addHours(dateFns.addDays(new Date(), 15), 1);
-            mockCertExpiryAndVisitHomepage(api.certExpiry.central, expiry);
+            const staticResponseMapGeneric = {
+                [credentialexpiryCentralAlias]: {
+                    body: { expiry },
+                },
+            };
+            visitForCredentialExpiry(staticResponseMapGeneric);
 
             cy.get(selectors.centralCertExpiryBanner).should('not.exist');
         });
 
         it('should display banner without download button if user does not have the required permission', () => {
-            cy.intercept('GET', api.permissions.mypermissions, {
-                body: {
-                    globalAccess: 'READ_ACCESS',
-                    resourceToAccess: {
-                        VulnerabilityManagementRequests: 'READ_ACCESS',
-                        VulnerabilityManagementApprovals: 'READ_ACCESS',
-                    },
-                },
-            }).as('permissions');
             const expiry = dateFns.addMinutes(dateFns.addHours(new Date(), 23), 30);
-            mockCertExpiryAndVisitHomepage(api.certExpiry.central, expiry);
-            cy.wait('@permissions');
+            const staticResponseMapGeneric = {
+                [mypermissionsAlias]: {
+                    fixture: 'auth/mypermissionsMinimalAccess.json',
+                },
+                [credentialexpiryCentralAlias]: {
+                    body: { expiry },
+                },
+            };
+            visitForCredentialExpiry(staticResponseMapGeneric);
```

Comment:
I guess considering this use case and my last comment, maybe using something like `visitMainDashboardWithCredentialExpiryResponse` as the function name would not be ideal because technically we're now mocking the `mypermissions` response too. Maybe a more generic function name would be better than to consider this and the previous use case

Reply:
Also, the alias naming seems a bit odd because we've been using this weird casing that's not quite camel case (`mypermissionsAlias` vs. `myPermissionsAlias`). I believe that stemmed from the backend naming convention. I guess that's something worth discussing with the backend team as well. Maybe we can try to limit them on these case styles so we don't have these weird situations